source/docs/hardware/sensors/encoders-hardware.rst

```diff
@@ -27,25 +27,32 @@ These encoders vary in how they are mounted to the mechanism in question.  In ad
 Shafted encoders
 ^^^^^^^^^^^^^^^^
 
-.. todo:: add picture
+.. image:: images/encoders-hardware/grayhill-63r.jpg
 
-Shafted encoders have a sealed body with a shaft protruding out of it that must be coupled rotationally to a mechanism.  This is often done with a helical beam coupling, or, more cheaply, with a length of flexible tubing (such as surgical tubing or pneumatic tubing), fastened with cable ties and/or adhesive at either end.  Many commercial off-the-shelf FRC gearboxes have purpose-built mounting points for shafted encoders, such as the popular `Grayhill 63r <http://www.grayhill.com/assets/1/7/Opt_Encoder_63R.pdf>`__.
+Shafted encoders have a sealed body with a shaft protruding out of it that must be coupled rotationally to a mechanism.  This is often done with a helical beam coupling, or, more cheaply, with a length of flexible tubing (such as surgical tubing or pneumatic tubing), fastened with cable ties and/or adhesive at either end.  Many commercial off-the-shelf FRC gearboxes have purpose-built mounting points for shafted encoders, such as the popular `Grayhill 63r <http://www.grayhill.com/assets/1/7/Opt_Encoder_63R.pdf>`__, pictured above.
 
 On-shaft encoders
 ^^^^^^^^^^^^^^^^^
 
-.. todo:: add picture
+.. image:: images/encoders-hardware/cui-amt102.png
```

Comment:
Is it possible to have a SVG picture here, it's a bit jarring. If not, it's fine.

Reply:
Added in 8330140.